src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php

```diff
@@ -39,11 +39,15 @@
      * Whatever value you return here will be passed to getUser() and checkCredentials()
      *
      * For example, for a form login, you might:
-     *
-     *      return array(
-     *          'username' => $request->request->get('_username'),
-     *          'password' => $request->request->get('_password'),
-     *      );
+     * 
+     *      if ($request->request->has('_username')) {
+     *          return array(
+     *              'username' => $request->request->get('_username'),
+     *              'password' => $request->request->get('_password'),
+     *          );
+     *      } else {
+     *          return;
+     *      }
```

Comment:
In the previous example, you return an array, even if no query is sent.

So when you arrive on your login/ form page, you send this array, so Guard tries to authenticate you.
As it fails to to it, you are redirected to... the login page again.
And so on.

As I said to @nicolas-grekas , the else is not necessary but the point is to show you should not forget this null case.